### ISSUE TITLE
Request - date display

### ISSUE THREAD

**Issue by andy4849845** [mpcjanssen/simpletask-android#287]
When displaying due dates, can they optionally be more sensible. Can they say today or tomorrow instead of the actual date, for instance?

**Comment by mpcjanssen** [comment]
This requires a more specialized date formatter than currently in the code. I might add this, but because of time constraints this is not going to happen any time soon.

**Comment by smichel17** [comment]
Seems like this is currently implemented when datebar is enabled. I think this issue could be closed if due dates got the same highlighting (red/green/none) whether or not the date bar is shown.

**Comment by smichel17** [comment]
#285, #173, and #175 also involve highlighting.

**Comment by kstep** [comment]
Also, it would be nice to have the ability to customize visible date format, like maybe an option to set it to `%a, %d %b %Y` or some other format string.

The often use case, is when I have to ask "oh, what exact date is "Wednesday"?" in conversation because SimpleTask doesn't show weekday names in dates, and I need to look up my schedule in my todo.txt file.

**Comment by smichel17** [comment]
I don't think it is wise to do custom formatting in-line, so I'm assuming we're talking about when the "Show datebar" setting is enabled.

Here's a list of things that Simpletask does or (checkboxes) could do for `days in future`:

- `[<0]`: "x [days|week|months|years] ago"
- `[0]` "today"
- [ ] `[1-6]` "[day-of-week]"
  - Today is Wednesday the 6th. Display the day of week up to & including Tuesday the 12th.
- [ ] `[7|14|21|28]` "in [1|2|3|4] [week|weeks]"
- [ ] `daysInMonth(TODAY)`: "in 1 month"
- [ ] Other dates `[8-30]`: "in [8-30] days"
- [ ] `[up to 364]` "[month abr.] [day-of-month] [st|nd|rd|th]"
- [ ] `[365]` "in 1 year"
- `[>365]` Precise/unformatted date.
